doc/task-hierarchy.md

```diff
@@ -0,0 +1,174 @@
+- Handling Commands
+- Author Identification
+- Sorting
+- Language Identification
+- Text Span Selection
+- Hallucination
+   - Given a context, generate imaginary content e.g. given a sentence, generate a story/poem.
+- Document Classification
+- Relevancy Estimation
+- Named Entity Recognition
+- Dialogue Understanding
+- Abuse Detection
+- Text Simplification
+- Tabular Text Operation
+- Coreference: Entity Coreference
+- Coreference: Pronoun Disambiguation
+- Reasoning: Logic
+- Reasoning: Numerical Reasoning
+- Reasoning: Reasoning on Actions
+- Reasoning: General Commonsense Reasoning
+- Reasoning: Abductive Reasoning
+- Reasoning: Temporal Reasoning
+- Reasoning: Spatial Reasoning
+- Reasoning: Multihop Reasoning
+- Reasoning: Reasoning on Social Interactions
+- Reasoning: Commonsense Reasoning
+   - Tasks related to activities humans do in daily life e.g. eating breakfast in the morning, sleeping during night etc.
+- Reasoning: Numerical Commonsense Reasoning
+   - Tasks which requires numerical commonsense knowledge e.g. a car has 4 wheels.
+- Reasoning: Physical Reasoning
+   - Tasks involving physical interactions with objects e.g. a knife (and not a paper) is used to cut objects
+- Reasoning: Logical Reasoning
+- Reasoning: Planning
+   - Tasks which need some sort of planning e.g. how to go to Hawaii?
+- Reasoning: Qualitative Reasoning
+- Reasoning: Deductive Reasoning
+- Reasoning: Causal Reasoning
+- Reasoning: Reasoning with Symbols
+   - Tasks where symbols represent various things e.g. if X is the number of apples in the freeze today morning and Y is the number remaining after I ate a few apples, X-Y is the number of apples I ate.
+
+Domain:
+* 9/11 reports
+* Abstract algebra
+* Accounting
+* Anatomy
+* Anthropology
+* Architecture
+* Art
+* Astronomy
+* Business ethics
+* Clinical knowledge
+* College biology
+* College chemistry
+* College computer science
+* College mathematics
+* College medicine
+* College physics
+* Computer security
+* Conceptual physics
+* Econometrics
+* Electrical engineering
+* Elementary mathematics
+* Fiction
+* Formal logic
+* Geography
+* Global facts
+* High school biology
+* High school chemistry
+* High school computer science
+* High school european history
+* High school geography
+* High school government and politics
+* High school macroeconomics
+* High school mathematics
+* High school microeconomics
+* High school physics
+* High school psychology
+* High school statistics
+* High school history
+* History
+* Human aging
+* Human sexuality
+* International law
+* Jurisprudence
+* Justice
+* Law
+* Logical fallacies
+* Machine learning
+* Management
+* Marketing
+* Medical genetics
+* Medicine
+* Moral scenarios
+* Movies
+* Music
+* News
+* Nutrition
+* Personal narratives
+* Philosophy
+* Prehistory
+* Psychology
+* Public relations
+* School science textbooks
+* Security  studies
```

Comment:
Not sure what kind of security this is. Is this computer security? national security?